storage/diemdb/src/transaction_store/mod.rs

```diff
@@ -128,8 +128,8 @@ impl TransactionStore {
     /// Returns None if there's no such transaction at or preceding `version` (it's likely the genesis
     /// version 0).
     pub fn get_block_metadata(&self, version: Version) -> Result<Option<(Version, BlockMetadata)>> {
-        // Maximum TPS from benchmark is around 1000.
-        const MAX_VERSIONS_TO_SEARCH: usize = 1000 * 3;
+        // this dictates the max block size
```

Comment:
// Must be larger than a block size, otherwise a NotFound error will be raised wrongly.